src/sendGCM.js

```diff
@@ -138,7 +138,7 @@ const sendGCM = (regIds, data, settings) => {
     restrictedPackageName: data.restrictedPackageName,
     dryRun: data.dryRun || false,
     data: opts.phonegap === true ? Object.assign(custom, notification) : custom, // See https://github.com/phonegap/phonegap-plugin-push/blob/master/docs/PAYLOAD.md#android-behaviour
-    notification: opts.phonegap === true ? undefined : notification,
+    notification: opts.phonegap === true || data.silent ? undefined : notification,
```

Comment:
please update README.md where currently the silent flag is declared as a apn only parameter.
Ideally add a subsection to the GCM section where the "data-only" notifications via silent=true are mentioned.